source/base/partitioner.cc

```diff
@@ -58,6 +58,28 @@ namespace Utilities
 
 
 
+    Partitioner::Partitioner(const unsigned int size,
+                             const unsigned int ghost_size,
+                             const MPI_Comm     communicator)
+      : global_size(
+          Utilities::MPI::sum<types::global_dof_index>(size, communicator))
+      , locally_owned_range_data(size)
+      , local_range_data(
+          std::pair<types::global_dof_index, types::global_dof_index>(0, size))
```

Comment:
This does not really correspond to the 1-to-1 and ascending indices I was expecting and what the line immediately above regarding `global_size` would suggest. Should this include the result of a prefix sum? I assume you do not really use the global indices or `Vector::operator()(types::global_dof_index)` functions because they would not have the right index.